src/type.js

```diff
@@ -4,44 +4,121 @@ import { core } from 'metal';
 
 const validators = {
 	any: () => true,
-	array: primitiveTypeValidatorFactory('array'),
-	bool: primitiveTypeValidatorFactory('boolean'),
-	func: primitiveTypeValidatorFactory('function'),
-	number: primitiveTypeValidatorFactory('number'),
-	object: primitiveTypeValidatorFactory('object'),
-	string: primitiveTypeValidatorFactory('string'),
-
-	arrayOf: arrayOfValidatorFactory,
-	instanceOf: instanceOfValidatorFactory,
-	objectOf: objectOfValidatorFactory,
-	oneOfType: oneOfTypeValidatorFactory,
-	shapeOf: shapeOfValidatorFactory
-};
-
-/**
- * Creates a validator that checks the values of an array against a type.
- * @param {!function} validator Type validator to check each index against.
- * @return {function} Validator.
- */
-function arrayOfValidatorFactory(validator) {
-	return (value, name, context) => {
-		if (!Array.isArray(value)) {
-			return composeError('Expected an array.', name, context);
-		} else {
-			const testArray = value.every(
-				item => {
-					return !(validator(item, name) instanceof Error);
+	array: validateType('array'),
+	bool: validateType('boolean'),
+	func: validateType('function'),
+	number: validateType('number'),
+	object: validateType('object'),
+	string: validateType('string'),
+
+	/**
+	 * Creates a validator that checks the values of an array against a type.
+	 * @param {!function} validator Type validator to check each index against.
```

Comment:
Oh good to know, I wasn't sure which standards to follow so I sort of guessed this time around. I can go through and fix the docs now though.